source/main/cardano/setupCardanoNodeMode.js

```diff
@@ -43,7 +44,11 @@ const startCardanoNode = (node: CardanoNode, launcherConfig: Object) => {
   return node.start(config);
 };
 
-const restartCardanoNode = async (node: CardanoNode) => {
+export const restartCardanoNode = async (node: CardanoNode) => {
+  if (
+    node.state === CardanoNodeStates.STARTING ||
+    node.state === CardanoNodeStates.RUNNING
```

Comment:
@daniloprates do we really want to prevent restart in case Cardano node is in `RUNNING` state? It looks to me we should allow the restart in this case and perhaps prevent it in other non-final states such as: `STOPPING, EXITING, UPDATING`...

Reply:
You're right, I moved the node.state checking into the `onCheckDiskSpace` function, as in this case it doesn't make sense to restart it if it's already running.